R/plotImage.R

```diff
@@ -31,6 +33,89 @@ NULL
 #' @export
 plotSpatialData <- \() ggplot() + scale_y_reverse() + .theme 
 
+# merge/manage image channels
+# if no colors and channels defined, return the first channel
+#' @noRd
+.manage_channels <- \(a, ch, c=NULL){
+  if(length(ch) > length(.DEFAULT_COLORS) && is.null(c))
+    stop("You can only choose at most seven default colors!")
+  if(!is.null(c) || (is.null(c) && length(ch) > 1)) {
+    if(is.null(c))
+      c <- .DEFAULT_COLORS[1:length(ch)] 
+    c <- col2rgb(c)/255
+    a_new <- array(0, dim = c(3,dim(a)[-1]))
+    for(i in 1:dim(a)[1]){
+      a_new[1,,] <-  a_new[1,,,drop = FALSE] + a[i,,,drop = FALSE]*c[1,i]
+      a_new[2,,] <-  a_new[2,,,drop = FALSE] + a[i,,,drop = FALSE]*c[2,i]
+      a_new[3,,] <-  a_new[3,,,drop = FALSE] + a[i,,,drop = FALSE]*c[3,i]
+    }
```

Comment:
this seems odd. `a_new` is all 0s, and we then add to it in a `for`-loop. Couldn't the same be done by just multiplying along `dim(a)[1]` by `c`? I.e., one line? I can give it a try, unless I'm missing something?